soroban-sdk/src/lib.rs

```diff
@@ -48,7 +48,7 @@
 compile_error!("'testutils' feature is not supported on 'wasm' target");
 
 // When used in a no_std contract, provide a panic handler as one is required.
-#[cfg(all(not(feature = "alloc"), target_family = "wasm"))]
+#[cfg(all(not(feature = "alloc"), not(feature = "std"), target_family = "wasm"))]
```

Comment:
This can be breaking for someone, can't it? No-one would have the new feature enabled, so `handle_panic` may disappear suddenly. Wouldn't it be safer to add a feature that disables `handle_panic` on-demand?